### ISSUE TITLE
Failed to initialize wallaby jest. "EBUSY: resource busy or locked" for Jest Haste-Map file

### ISSUE THREAD

**Issue by dggsax** [wallabyjs/public#2741]
### Issue description or question

When running Wallaby.js with automatic configuration I receive the following error:

```
Failed to initialize wallaby jest. 
EBUSY: resource busy or locked, open 'C:\Users\FAKEUSER\AppData\Local\Temp\jest\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8' 
```

It feels like it's a synchronous issue between the workers threads? I've tried to track the locks on the generated `haste-map-*` file, but can't seem to get to the bottom of what's going on here. The source for my project is located on my `D:` drive and the file being locked is in the `C:` drive. I tried to override the `caceDirectory` configuration, but was unable to do so.

Any help is greatly appreciated!

### Wallaby diagnostics report

I saw this mentioned in another issue, but it would be great if the generated diagnostic report anonymized at the very least the user name and the paths to the project. I went ahead and anonymized by hand, 

```javascript
{
  editorVersion: '1.57.1',
  pluginVersion: '1.0.297',
  editorType: 'VSCode',
  osVersion: 'win32 10.0.19043',
  nodeVersion: 'v14.15.0',
  coreVersion: '1.0.1098',
  checksum: 'NWI5MTQwZTAwY2Q2ZTA2MDBlZjIxZmY0MTU5NGIxZmIsMTYzOTc4NTYwMDAwMCww',
  config: {
    workers: { initial: 10, regular: 5, recycle: false },
    diagnostics: {
      jest: {
        config: {
          configs: [
            {
              automock: false,
              cache: true,
              cacheDirectory: 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest',
              clearMocks: false,
              coveragePathIgnorePatterns: [
                'D:\\\\FAKEPROJECTDIRECTORY\\\\src\\\\serviceWorker.ts',
                'D:\\\\FAKEPROJECTDIRECTORY\\\\src\\\\setupTests.ts',
                'D:\\\\FAKEPROJECTDIRECTORY\\\\src\\\\utils\\\\test-utils.tsx',
                'D:\\\\FAKEPROJECTDIRECTORY\\\\src\\\\models'
              ],
              cwd: 'D:\\FAKEPROJECTDIRECTORY',
              dependencyExtractor: undefined,
              detectLeaks: undefined,
              detectOpenHandles: undefined,
              displayName: undefined,
              errorOnDeprecated: false,
              extraGlobals: [],
              filter: undefined,
              forceCoverageMatch: [],
              globalSetup: undefined,
              globalTeardown: undefined,
              globals: {},
              haste: { computeSha1: false, throwOnModuleCollision: false },
              injectGlobals: true,
              moduleDirectories: [ 'node_modules' ],
              moduleFileExtensions: [
                'web.js',  'js',
                'web.ts',  'ts',
                'web.tsx', 'tsx',
                'json',    'web.jsx',
                'jsx',     'node'
              ],
              moduleLoader: undefined,
              moduleNameMapper: [ [ '^react-native$', 'react-native-web' ], [ '^.+\\.module\\.(css|sass|scss)$', 'identity-obj-proxy' ] ],
              modulePathIgnorePatterns: [],
              modulePaths: [],
              name: 'e65d8e6931b212ff283a9131d2dde0c1',
              prettierPath: 'prettier',
              resetMocks: true,
              resetModules: false,
              resolver: undefined,
              restoreMocks: false,
              rootDir: 'D:\\FAKEPROJECTDIRECTORY',
              roots: [ 'D:\\FAKEPROJECTDIRECTORY\\src' ],
              runner: 'jest-runner',
              setupFiles: [ 'D:\\FAKEPROJECTDIRECTORY\\node_modules\\react-app-polyfill\\jsdom.js' ],
              setupFilesAfterEnv: [ 'D:\\FAKEPROJECTDIRECTORY\\src\\setupTests.ts' ],
              skipFilter: false,
              skipNodeResolution: undefined,
              slowTestThreshold: 5,
              snapshotResolver: undefined,
              snapshotSerializers: [],
              testEnvironment: 'D:\\FAKEPROJECTDIRECTORY\\node_modules\\jest-environment-jsdom\\build\\index.js',
              testEnvironmentOptions: {},
              testLocationInResults: false,
              testMatch: [
                'D:/FAKEPROJECTDIRECTORY/src/**/__tests__/**/*.{js,jsx,ts,tsx}',
                'D:/FAKEPROJECTDIRECTORY/src/**/*.{spec,test}.{js,jsx,ts,tsx}'
              ],
              testPathIgnorePatterns: [ '\\\\node_modules\\\\' ],
              testRegex: [],
              testRunner: 'D:\\FAKEPROJECTDIRECTORY\\node_modules\\jest-circus\\runner.js',
              testURL: 'http://localhost',
              timers: 'real',
              transform: [
                [
                  '^.+\\.(js|jsx|mjs|cjs|ts|tsx)$',
                  'D:\\FAKEPROJECTDIRECTORY\\node_modules\\react-scripts\\config\\jest\\babelTransform.js',
                  {}
                ],
                [ '^.+\\.css$', 'D:\\FAKEPROJECTDIRECTORY\\node_modules\\react-scripts\\config\\jest\\cssTransform.js', {} ],
                [
                  '^(?!.*\\.(js|jsx|mjs|cjs|ts|tsx|css|json)$)',
                  'D:\\FAKEPROJECTDIRECTORY\\node_modules\\react-scripts\\config\\jest\\fileTransform.js',
                  {}
                ]
              ],
              transformIgnorePatterns: [ '[\\\\\\\\]node_modules[\\\\\\\\].+\\.(js|jsx|mjs|cjs|ts|tsx)$', '^.+\\.module\\.(css|sass|scss)$' ],
              unmockedModulePathPatterns: undefined,
              watchPathIgnorePatterns: []
            }
          ],
          globalConfig: {
            bail: 0,
            changedFilesWithAncestor: false,
            changedSince: undefined,
            collectCoverage: false,
            collectCoverageFrom: [ 'src/**/*.{js,jsx,ts,tsx}', '!src/**/*.d.ts' ],
            collectCoverageOnlyFrom: undefined,
            coverageDirectory: 'D:\\FAKEPROJECTDIRECTORY\\coverage',
            coverageProvider: 'babel',
            coverageReporters: [ 'json', 'text', 'lcov', 'clover' ],
            coverageThreshold: undefined,
            detectLeaks: undefined,
            detectOpenHandles: undefined,
            enabledTestsMap: undefined,
            errorOnDeprecated: false,
            expand: false,
            filter: undefined,
            findRelatedTests: false,
            forceExit: false,
            globalSetup: undefined,
            globalTeardown: undefined,
            json: false,
            lastCommit: false,
            listTests: undefined,
            logHeapUsage: false,
            maxConcurrency: 5,
            maxWorkers: 15,
            noSCM: undefined,
            noStackTrace: false,
            nonFlagArgs: undefined,
            notify: false,
            notifyMode: 'failure-change',
            onlyChanged: false,
            onlyFailures: false,
            outputFile: undefined,
            passWithNoTests: undefined,
            projects: [],
            replname: undefined,
            reporters: undefined,
            rootDir: 'D:\\FAKEPROJECTDIRECTORY',
            runTestsByPath: false,
            silent: undefined,
            skipFilter: false,
            testFailureExitCode: 1,
            testNamePattern: undefined,
            testPathPattern: '',
            testResultsProcessor: undefined,
            testSequencer: 'D:\\FAKEPROJECTDIRECTORY\\node_modules\\@jest\\test-sequencer\\build\\index.js',
            testTimeout: undefined,
            updateSnapshot: 'new',
            useStderr: false,
            verbose: undefined,
            watch: false,
            watchAll: false,
            watchPlugins: [
              { config: {}, path: 'D:\\FAKEPROJECTDIRECTORY\\node_modules\\jest-watch-typeahead\\filename.js' },
              { config: {}, path: 'D:\\FAKEPROJECTDIRECTORY\\node_modules\\jest-watch-typeahead\\testname.js' }
            ],
            watchman: true
          },
          hasDeprecationWarnings: false,
          wallaby: {
            roots: [ 'src' ],
            watchPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\.\\\\git\\\\|\\\\\\..+\\\\' ],
            testPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\.\\\\git\\\\|\\\\\\..+\\\\' ],
            testMatch: [ 'src/**/__tests__/**/*.{js,jsx,ts,tsx}', 'src/**/*.{spec,test}.{js,jsx,ts,tsx}' ],
            testRegex: []
          }
        }
      }
    },
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', autoDetected: true },
    filesWithCoverageCalculated: [ 'src/**/*.{js,jsx,ts,tsx}', '!src/**/*.d.ts' ],
    filesWithNoCoverageCalculated: [
      /D:\\FAKEPROJECTDIRECTORY\\src\\serviceWorker.ts/,
      /D:\\FAKEPROJECTDIRECTORY\\src\\setupTests.ts/,
      /D:\\FAKEPROJECTDIRECTORY\\src\\utils\\test-utils.tsx/,
      /D:\\FAKEPROJECTDIRECTORY\\src\\models/
    ],
    globalSetup: false,
    micromatch: true,
    files: [
      { pattern: '\\\\node_modules\\\\', regexp: /\\node_modules\\/, ignore: true, trigger: true, load: true },
      {
        pattern: '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\.\\\\git\\\\|\\\\\\..+\\\\',
        regexp: /\.\\dist\\|\.\\build\\|\.\\coverage\\|\.\\git\\|\\\..+\\/,
        ignore: true,
        trigger: true,
        load: true
      },
      { pattern: '**/**', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: 'src/**', ignore: false, trigger: true, load: true, order: 2 },
      { pattern: 'src/**/__tests__/**/*.{js,jsx,ts,tsx}', ignore: true, trigger: true, load: true },
      { pattern: 'src/**/*.{spec,test}.{js,jsx,ts,tsx}', ignore: true, trigger: true, load: true }
    ],
    tests: [
      { pattern: '\\\\node_modules\\\\', regexp: /\\node_modules\\/, ignore: true, trigger: true, load: true, test: true },
      {
        pattern: '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\.\\\\git\\\\|\\\\\\..+\\\\',
        regexp: /\.\\dist\\|\.\\build\\|\.\\coverage\\|\.\\git\\|\\\..+\\/,
        ignore: true,
        trigger: true,
        load: true,
        test: true
      },
      { pattern: 'src/**/__tests__/**/*.{js,jsx,ts,tsx}', ignore: false, trigger: true, load: true, test: true, order: 3 },
      { pattern: 'src/**/*.{spec,test}.{js,jsx,ts,tsx}', ignore: false, trigger: true, load: true, test: true, order: 4 }
    ],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    addModifiedTestFileToExclusiveTestRun: true,
    compilers: {},
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    mapConsoleMessagesStackTrace: false,
    extensions: {},
    env: { type: 'node', params: {}, runner: 'C:\\Program Files\\nodejs\\node.exe', viewportSize: { width: 800, height: 600 }, options: { width: 800, height: 600 }, bundle: true },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    loose: true,
    configCode: 'auto.detect#-2117749550'
  },
  packageJSON: {
    dependencies: {
      '@azure/msal-browser': '^2.14.2',
      '@fluentui/react': '^8.19.1',
      '@fluentui/react-charting': '^5.2.10',
      '@reduxjs/toolkit': '^1.6.0',
      '@types/lodash': '^4.14.170',
      '@types/node': '^15.12.2',
      '@types/react': '^17.0.11',
      '@types/react-csv': '^1.1.1',
      '@types/react-dom': '^17.0.7',
      '@types/react-redux': '^7.1.16',
      '@types/react-router-dom': '^5.1.7',
      '@types/react-syntax-highlighter': '^13.5.1',
      lodash: '^4.17.21',
      msal: '^1.4.11',
      'powerbi-client-react': '^1.3.3',
      react: '^17.0.2',
      'react-aad-msal': '^2.3.5',
      'react-csv': '^2.0.3',
      'react-dom': '^17.0.2',
      'react-markdown': '^6.0.2',
      'react-redux': '^7.2.4',
      'react-router-dom': '^5.2.0',
      'react-scripts': '4.0.3',
      'react-syntax-highlighter': '^15.4.3',
      redux: '^4.1.0',
      'redux-thunk': '^2.3.0',
      runtypes: '^6.3.1',
      typescript: '^4.3.4',
      uuid: '^8.3.2'
    },
    devDependencies: {
      '@testing-library/dom': '^7.31.2',
      '@testing-library/jest-dom': '^5.14.1',
      '@testing-library/react': '^11.2.7',
      '@testing-library/react-hooks': '^7.0.0',
      '@testing-library/user-event': '^13.1.9',
      '@types/deep-freeze': '^0.1.2',
      '@types/enzyme': '3.10.8',
      '@types/enzyme-adapter-react-16': '1.0.6',
      '@types/fetch-mock': '^7.3.3',
      '@types/jest': '^26.0.23',
      '@types/react-test-renderer': '^17.0.1',
      '@types/redux-immutable-state-invariant': '^2.1.1',
      '@types/redux-logger': '^3.0.8',
      '@types/redux-mock-store': '^1.0.2',
      '@typescript-eslint/eslint-plugin': '^4.27.0',
      '@typescript-eslint/parser': '^4.27.0',
      'deep-freeze': '0.0.1',
      enzyme: '3.11.0',
      'enzyme-adapter-react-16': '1.15.6',
      'enzyme-to-json': '3.6.2',
      'eslint-plugin-simple-import-sort': '^7.0.0',
      'fetch-mock': '^9.11.0',
      'jest-cli': '^27.0.5',
      'jest-fetch-mock': '3.0.3',
      msw: '^0.29.0',
      prettier: '^2.3.1',
      'react-test-renderer': '^17.0.2',
      'redux-immutable-state-invariant': '^2.1.0',
      'redux-mock-store': '^1.5.4',
      'ts-jest': '27.0.3'
    }
  },
  fs: { numberOfFiles: 142 },
  debug: [
    '2021-06-26T22:18:49.191Z angular/cli config Angular CLI not found.\n',
    '2021-06-26T22:18:49.644Z jest/config Detected Jest.\n',
    '2021-06-26T22:18:49.644Z jest/config Configured Jest.\n',
    '2021-06-26T22:18:49.646Z project Wallaby Node version: v14.15.0\n',
    '2021-06-26T22:18:49.646Z project Wallaby config: D:\\FAKEPROJECTDIRECTORY\\auto.detect\n',
    '2021-06-26T22:18:50.084Z project File cache: C:\\Users\\FAKEUSERNAME\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.297\\projects\\34141ee7fa58a8c0\n',
    '2021-06-26T22:18:50.285Z uiService Listening port 51235\n',
    '2021-06-26T22:18:50.325Z project package.json file change detected, invalidating local cache\n',
    '2021-06-26T22:18:50.365Z workers Parallelism for initial run: 10, for regular run: 5\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #0\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #1\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #2\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #3\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #4\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #5\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #6\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #7\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #8\n',
    '2021-06-26T22:18:50.366Z workers Starting run worker instance #9\n',
    '2021-06-26T22:18:50.367Z workers Web server is listening at 63589\n',
    '2021-06-26T22:18:50.367Z project File cache requires some updates, waiting required files from IDE\n',
    '2021-06-26T22:18:50.590Z project Stopping process pool\n',
    '2021-06-26T22:18:50.591Z project Test run started; run priority: 3\n',
    '2021-06-26T22:18:50.593Z project Running all tests\n',
    '2021-06-26T22:18:50.597Z workers Starting test run, priority: 3\n',
    '2021-06-26T22:18:50.597Z workers Distributing tests between 10 workers\n',
    '2021-06-26T22:18:50.598Z workers Running tests in parallel\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #0, session #gc8m5]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #1, session #o27jx]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #2, session #ibuvl]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #3, session #0exl2]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #4, session #y7wfz]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #5, session #b1z7a]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #6, session #5fkje]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #7, session #hpuqs]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #8, session #npcwr]\n',
    '2021-06-26T22:18:50.599Z nodeRunner Starting sandbox [worker #9, session #3q56n]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #0, session #gc8m5]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #1, session #o27jx]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #2, session #ibuvl]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #3, session #0exl2]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #4, session #y7wfz]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #5, session #b1z7a]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #6, session #5fkje]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #7, session #hpuqs]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #8, session #npcwr]\n',
    '2021-06-26T22:18:50.600Z nodeRunner Preparing sandbox [worker #9, session #3q56n]\n',
    '2021-06-26T22:18:50.636Z workers Started run worker instance (delayed) #0\n',
    '2021-06-26T22:18:50.636Z nodeRunner Prepared sandbox [worker #0, session #gc8m5]\n',
    '2021-06-26T22:18:50.636Z workers [worker #0, session #gc8m5] Running tests in sandbox\n',
    '2021-06-26T22:18:50.640Z workers Started run worker instance (delayed) #1\n',
    '2021-06-26T22:18:50.640Z nodeRunner Prepared sandbox [worker #1, session #o27jx]\n',
    '2021-06-26T22:18:50.640Z workers [worker #1, session #o27jx] Running tests in sandbox\n',
    '2021-06-26T22:18:50.648Z workers Started run worker instance (delayed) #2\n',
    '2021-06-26T22:18:50.648Z nodeRunner Prepared sandbox [worker #2, session #ibuvl]\n',
    '2021-06-26T22:18:50.648Z workers [worker #2, session #ibuvl] Running tests in sandbox\n',
    '2021-06-26T22:18:50.653Z workers Started run worker instance (delayed) #3\n',
    '2021-06-26T22:18:50.653Z nodeRunner Prepared sandbox [worker #3, session #0exl2]\n',
    '2021-06-26T22:18:50.653Z workers [worker #3, session #0exl2] Running tests in sandbox\n',
    '2021-06-26T22:18:50.658Z workers Started run worker instance (delayed) #5\n',
    '2021-06-26T22:18:50.659Z nodeRunner Prepared sandbox [worker #5, session #b1z7a]\n',
    '2021-06-26T22:18:50.659Z workers [worker #5, session #b1z7a] Running tests in sandbox\n',
    '2021-06-26T22:18:50.664Z workers Started run worker instance (delayed) #4\n',
    '2021-06-26T22:18:50.665Z nodeRunner Prepared sandbox [worker #4, session #y7wfz]\n',
    '2021-06-26T22:18:50.665Z workers [worker #4, session #y7wfz] Running tests in sandbox\n',
    '2021-06-26T22:18:50.669Z workers Started run worker instance (delayed) #6\n',
    '2021-06-26T22:18:50.669Z nodeRunner Prepared sandbox [worker #6, session #5fkje]\n',
    '2021-06-26T22:18:50.669Z workers [worker #6, session #5fkje] Running tests in sandbox\n',
    '2021-06-26T22:18:50.672Z workers Started run worker instance (delayed) #8\n',
    '2021-06-26T22:18:50.672Z nodeRunner Prepared sandbox [worker #8, session #npcwr]\n',
    '2021-06-26T22:18:50.672Z workers [worker #8, session #npcwr] Running tests in sandbox\n',
    '2021-06-26T22:18:50.677Z workers Started run worker instance (delayed) #9\n',
    '2021-06-26T22:18:50.677Z nodeRunner Prepared sandbox [worker #9, session #3q56n]\n',
    '2021-06-26T22:18:50.677Z workers [worker #9, session #3q56n] Running tests in sandbox\n',
    '2021-06-26T22:18:50.678Z workers Started run worker instance (delayed) #7\n',
    '2021-06-26T22:18:50.678Z nodeRunner Prepared sandbox [worker #7, session #hpuqs]\n',
    '2021-06-26T22:18:50.678Z workers [worker #7, session #hpuqs] Running tests in sandbox\n',
    '2021-06-26T22:18:52.582Z workers Sandbox (active) [o27jx] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.593Z workers Sandbox (active) [0exl2] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.613Z workers Sandbox (active) [b1z7a] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.613Z workers Sandbox (active) [y7wfz] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.622Z workers Sandbox (active) [gc8m5] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.634Z workers Sandbox (active) [npcwr] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.638Z workers Sandbox (active) [hpuqs] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.643Z workers Sandbox (active) [5fkje] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.644Z workers Sandbox (active) [3q56n] error: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.687Z workers [o27jx] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.690Z workers [o27jx] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.691Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.695Z workers [0exl2] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.697Z workers [0exl2] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.698Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.718Z workers [b1z7a] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.718Z workers [y7wfz] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.721Z workers [gc8m5] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.722Z workers [b1z7a] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.722Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.723Z workers [y7wfz] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.723Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.725Z workers [gc8m5] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.725Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.749Z workers [npcwr] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.749Z workers [hpuqs] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.749Z workers [5fkje] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.749Z workers [3q56n] Run 0 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:18:52.752Z workers [npcwr] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.753Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.754Z workers [hpuqs] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.754Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.755Z workers [5fkje] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.755Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:18:52.755Z workers [3q56n] Sandbox is responsive, closing it\n',
    '2021-06-26T22:18:52.755Z workers Failed to map the stack to user code, entry message: Failed to initialize wallaby jest.\n' +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8', stack: Failed to initialize wallaby jest.\n" +
      "EBUSY: resource busy or locked, open 'C:\\Users\\FAKEUSERNAME\\AppData\\Local\\Temp\\jest\\haste-map-e65d8e6931b212ff283a9131d2dde0c1-c2017e33bf8fbce627f3a9eaa61653d3-da9dff6bcf050bae16bdd1188e0061b8'\n",
    '2021-06-26T22:19:14.962Z workers Scheduling Jest Test Run (ibuvl): 2021-06-26T22:18:53.076Z\n',
    '2021-06-26T22:19:15.029Z workers [ibuvl] Loaded unknown number of test(s)\n',
    '2021-06-26T22:19:15.029Z workers [ibuvl] Test executed: render\n',
    '2021-06-26T22:19:15.031Z workers [ibuvl] Loaded unknown number of test(s)\n',
    '2021-06-26T22:19:15.031Z workers [ibuvl] Test executed: responds to resize instantly\n',
    '2021-06-26T22:19:16.554Z workers [ibuvl] Test executed: handles delay when timeout parameter is set\n',
    '2021-06-26T22:19:16.569Z workers [ibuvl] Run 3 test(s), skipped 0 test(s)\n',
    '2021-06-26T22:19:16.569Z workers Jest Test Run Complete (ibuvl): 2021-06-26T22:19:16.549Z\n',
    '2021-06-26T22:19:16.572Z workers [ibuvl] Sandbox is responsive, closing it\n',
    '2021-06-26T22:19:16.573Z workers Merging parallel test run results\n',
    '2021-06-26T22:19:16.574Z project Test run finished\n',
    '2021-06-26T22:19:16.576Z project Processed console.log entries\n',
    '2021-06-26T22:19:16.576Z project Processed loading sequences\n',
    '2021-06-26T22:19:16.577Z project Processed executed tests\n',
    '2021-06-26T22:19:16.585Z project Processed code coverage\n',
    '2021-06-26T22:19:16.612Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by smcenlly** [comment]
> It feels like it's a synchronous issue between the workers threads? I've tried to track the locks on the generated `haste-map-*` file, but can't seem to get to the bottom of what's going on here. 

Have you tried rebooting your machine since receiving the error? It's possible that the file is being locked by a process other than Wallaby. 

You may try and test your theory about being a multiple workers issue by changing [Wallaby's `workers` setting](https://wallabyjs.com/docs/config/workers.html) and set both `workers.initial` and `workers.regular` to `1`. For example:

**wallaby.js (configuration file)**
```javascript
module.exports = ({
  autoDetect: true,
  workers: {
    initial: 1,
    regular: 1
  }
});
```

> The source for my project is located on my `D:` drive and the file being locked is in the `C:` drive.

We tested a project located on `D:` and cache directory on `C:` and everything appeared to work for us. We don't believe this is a problem.

> I tried to override the `caceDirectory` configuration, but was unable to do so.

`caceDirectory` is incorrectly spelled in your message, is it possible that it was also incorrectly spelled in your jest configuration? Wallaby overrides the cache directory if it is specified in your jest config. For example:

**jest.config.js**
```javascript
module.exports = {
  cacheDirectory: 'c:\\temp\\jest_cache',
};
```

> I saw this mentioned in another issue, but it would be great if the generated diagnostic report anonymized at the very least the user name and the paths to the project. I went ahead and anonymized by hand, 

I agree we could do something to improve anonymizing the diagnostic report by replacing `username` but sometimes paths are important so would not able to adjust that. Regardless of what efforts we went to anonymize the diagnostics report, it should always be reviewed and we want to be very careful to ensure that people don't assume the diagnostics report is already anonymized and then send it on without proper review/sanitization. We've created a feature request to track any changes we make to anonymizing the diagnostics report  (#2742).

**Comment by dggsax** [comment]
@smcenlly - thank you for the prompt response!

**Thread-safe theory**
I did what you suggested and ran with the configuration to make it so that it used 1 worker, and all 112 of my tests were able to execute and run. I incremented the `initial` and `regular` numbers until I saw the issue re-appear, and it re-appeared with just 3 workers.

As for other processes on my machine locking the file, I don't believe that's happening as I'm able to delete the file before/after wallaby execution, if another process were keeping a lock on the file, Windows wouldn't let me delete it.

I deleted the generated `haste-map-*` file, re-started Wallaby with `initial` & `regular` set to 3 on the config, and the file was immediately re-created, and the console gave me the same message about the lock contention.

I don't know how Wallaby works under the hood and can't poke around in the code, would this be an issue with the Wallaby engine itself or Jest? Could help to figure out what is accessing the file in a non-thread-safe way.

**Disk theory**
I don't have a `jest.config.js` file in my CRA project, the only jest settings I have specified is the `coveragePathIgnorePatterns` setting. If I wanted to test my disk theory, how would I specify the `cacheDirectory` setting in my wallaby config?

Let me know how I can help or what I can do to provide more information 😀

**Comment by ArtemGovorov** [comment]
@dggsax Can you please try disabling your anti-malware/antivirus/defender software and try again (with 3 or more workers)?

**Comment by dggsax** [comment]
@ArtemGovorov I am unable to disable anti-virus because my company does not allow that. I did go ahead and add the `C:\Users\FAKEUSER\AppData\Local\Temp\jest\` directory to defender exclusions which helped when specifying the number of workers to like 5, but when I increase the value of workers, or which my wallaby instance to us automatic configuration, I see the issue pop up again.

**Comment by smcenlly** [comment]
@dggsax - it seems unlikely that adding the `C:\Users\FAKEUSER\AppData\Local\Temp\jest\` directory would work with 5 works but not more. Did you try removing the exclusion? If so, did `5` workers still run OK?

Given that Jest is caching the results, you may be able to gradually increase the number of workers by 1 as the number of concurrent workers **writing** to the cache at the same time should only be the latest new worker (all others should only be performing a **read**). We haven't seen this problem before and still think it may be related to your specific operating system (e.g. virus scanner, etc.).

We have added some additional logging to Wallaby to help us isolate which parts of jest are causing the problem for you. We may be able to patch that code when running in Wallaby to retry on read failure. Could you please download the latest version of Wallaby core, `v1.0.1099` and provide us with an updated diagnostics report?

---

Are you able to run the same project on another machine where you are able to completely disable the virus scanner? If so, do you get the same problem?

**Comment by dggsax** [comment]
Hi @smcenlly  sorry for the radio silence, it's been a crazy week with work so haven't had the chance to try the latest guidance from you. I'll be sure to give it a go tomorrow and update you with how it goes, thank you!

**Comment by smcenlly** [comment]
@dggsax - did you have a chance to try this this again?

**Comment by smcenlly** [comment]
Closing this issue as we have not heard back. Will re-open if/when we get a reply.

**Comment by WaldiirNeto** [comment]
I know it's an old topic, but I had this problem today, I realized that I had enabled the option in vscode to run unit tests automatically.. so when I run Jest in the terminal I created two test instances at the same time and it was giving this error, I solved it by disabling the vscode tests option
